src/main/java/org/zalando/nakadi/partitioning/StringHash.java

```diff
@@ -0,0 +1,18 @@
+package org.zalando.nakadi.partitioning;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class StringHash {
+
+    public int hashCode(final String string) {
```

Comment:
@adyach you're right, I think we cannot change it since that would change how partitioning is working right now. This might not be the most sophisticated hashing out there, but there are some probability checks that show this hashing spreads quite well https://github.com/zalando/nakadi/blob/4b771e748af3cce5b8ff7519d28e1461d51d64a1/src/test/java/org/zalando/nakadi/partitioning/HashPartitionStrategyTest.java#L63-L76

Another thing I realised is that it overflows for strings bigger than 32 chars in length. But it seems that losing the bytes that don't fit is not an issue.